nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -153,6 +156,10 @@ let
           poolImported "${pool}" || poolImport "${pool}"  # Try one last time, e.g. to import a degraded pool.
         fi
         if poolImported "${pool}"; then
+        ${let cmd = concatMapStringsSep "\n" (elem: "clevis decrypt < /etc/clevis/${elem}.jwe | zfs load-key ${elem}") (filter (p: (elemAt (splitString "/" p) 0) == pool) clevisDatasets); in
+            optionalString (elem pool (map (e: elemAt (splitString "/" e) 0) clevisDatasets)) (optionalString (cmd != "") (cmd + " || "))}
```

Comment:
`elem pool (map (e: elemAt (splitString "/" e) 0) clevisDatasets)`

This condition seems redundant. Shouldn't it always be the same as `cmd != ""`, because of the `filter` in `cmd`'s definition?

Reply:
Agreed